examples/tic-tac-toe/ui/src/hooks/useTrophyQuery.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useSuiClient } from '@mysten/dapp-kit';
+import { normalizeSuiAddress } from '@mysten/sui/utils';
+import { useQuery, useQueryClient, UseQueryResult } from '@tanstack/react-query';
+import { Game } from 'hooks/useGameQuery';
+import { useTransactions } from 'hooks/useTransactions';
+
+export enum Trophy {
+	None = 0,
+	Draw,
+	Win,
+}
+
+export type UseTrophyQueryResponse = UseQueryResult<Trophy, Error>;
+export type InvalidateTrophyQuery = () => void;
+
+/** Refetch trophy status every 5 seconds */
+const REFETCH_INTERVAL = 5000;
+
+/**
+ * Query the trophy state of the game (whether the game has a winner
+ * or not).
+ *
+ * `id` is the Object ID of the game, and `kind` is what kind of game
+ * it is (whether it is shared or owned). The query in this hook
+ * depends on the value of `kind` (will not be enabled unless `kind`
+ * is available).
+ */
+export function useTrophyQuery(game?: Game): [UseTrophyQueryResponse, InvalidateTrophy] {
+	const client = useSuiClient();
+	const queryClient = useQueryClient();
+	const tx = useTransactions()!!;
+
+	const response = useQuery({
+		enabled: !!game,
+		refetchInterval: REFETCH_INTERVAL,
+		queryKey: ['game-end-state', game?.id],
```

Comment:
What might be convenient is to export all `queryKey` names so you could theoretically invalidate keys more generic throughout the app, if you needed to. E.g. `QueryKeys { GameEndState, ...}`, instead of having to expose an "invalidate" function on each of these.

Returning the "useQuery" hook as is, also exports a `refetch` function that you can use to re-fetch the values, instead of re-defining a new `invalidate` function on every hook.

Reply:
I thought about this, but I think again it kind of runs counter to the way `dapp-kit` is working, which abstracts away the query keys in its queries. Maybe if we can align on that, this would be a good way to go.